src/cunumeric/index/zip.cc

```diff
@@ -43,14 +44,25 @@ struct ZipImplBody<VariantKind::CPU, DIM, N> {
         auto outptr                       = out.ptr(rect);
         for (size_t idx = 0; idx < volume; ++idx) {
           Legion::Point<N> new_point;
-          for (size_t i = 0; i < N; i++) { new_point[i] = indx_ptrs[i][idx]; }
+          for (size_t i = 0; i < N; i++) {
+            if (indx_ptrs[i][idx] < 0) {
+              new_point[i] = shape[i] + indx_ptrs[i][idx];
+            } else {
+              new_point[i] = indx_ptrs[i][idx];
+            }
```

Comment:
I would rather do something like this (in this and other files), which does only one access on each array:
```
new_point[i] = wrap(indx_ptrs[i][idx], shape[i]);
```
where:
```
constexpr coord_t wrap(coord_t index, coord_t dim) {
  return index < 0 ? index + dim : index;
}
```